apps/api/src/app/layouts/use-cases/create-layout-change/create-layout-change.use-case.ts

```diff
@@ -0,0 +1,33 @@
+import { LayoutEntity, LayoutRepository } from '@novu/dal';
+import { ChangeEntityTypeEnum } from '@novu/shared';
+import { Injectable } from '@nestjs/common';
+
+import { CreateLayoutChangeCommand } from './create-layout-change.command';
+
+import { LayoutDto } from '../../dtos/layout.dto';
+import { ChannelTypeEnum, IEmailBlock } from '../../types';
+import { CreateChange, CreateChangeCommand } from '../../../change/usecases';
+
+@Injectable()
+export class CreateLayoutChangeUseCase {
+  constructor(private createChange: CreateChange, private layoutRepository: LayoutRepository) {}
+
+  async execute(command: CreateLayoutChangeCommand): Promise<void> {
+    const item = await this.layoutRepository.findById(command.layoutId, command.environmentId);
+
+    if (item) {
+      const changeId = LayoutRepository.createObjectId();
+
+      await this.createChange.execute(
+        CreateChangeCommand.create({
+          organizationId: command.organizationId,
+          environmentId: command.environmentId,
+          userId: command.userId,
+          type: ChangeEntityTypeEnum.LAYOUT,
+          item,
+          changeId,
+        })
+      );
+    }
+  }
```

Comment:
To review. I assume we don't need any parentChangeId as the layout changes are not based on another change.

Reply:
I agree :)